src/bicep/modules/SentinelSolution.bicep

```diff
@@ -0,0 +1,20 @@
+param workspaceName  string
+param workspaceId string
+param location string = resourceGroup().location
+param tags object = {}
+
+resource azurerm_log_analytics_solution 'Microsoft.OperationsManagement/solutions@2015-11-01-preview'={
+  name: 'SecurityInsights(${workspaceName})'
+  location: location
+  tags:tags
+  
+  properties: {
+    workspaceResourceId: workspaceId
```

Comment:
A common point of confusion when provisioning resources against a Log Analytics Workspace is the use of `workspace ID` and `Log Analytics Workspace Resource ID` -- I want to say this is the Resource ID of the Log Analytics Workspace resource and not the ID of the workspace, correct?

Instead, let's use pull in the Log Analytics Workspace resource into this module with an `existing` resource and refer to the ID with the implied dependency. Something like:

```bicep
resource workspace 'Microsoft.OperationalInsights/workspaces@2020-08-01' existing = {
  name: workspaceName
}

resource sentinelSolution 'Microsoft.OperationsManagement/solutions@2015-11-01-preview'={
  ...
  properties: {
    workspaceResourceId: workspace.id
  }
  ...
}
```

This implied dependency (`workspace.<foo>`) has the benefit of ensuring the resource exists before attempting to create the Sentinel solution.